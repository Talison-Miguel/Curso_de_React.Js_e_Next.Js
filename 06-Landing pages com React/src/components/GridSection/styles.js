import styled, { css } from 'styled-components';

export const Container = styled.div`
    ${({ theme }) => css``}
`;

export const Grid = styled.div`
    ${({ theme }) => css``}
`;

export const GridElement = styled.div`
    ${({ theme }) => css``}
`;
