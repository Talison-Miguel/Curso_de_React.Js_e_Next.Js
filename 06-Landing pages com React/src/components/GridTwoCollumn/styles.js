import styled, { css } from 'styled-components';

export const Container = styled.div`
    ${({ theme }) => css``}
`;

export const TextContainer = styled.div`
    ${({ theme }) => css``}
`;

export const ImageContainer = styled.div`
    ${({ theme }) => css``}
`;
