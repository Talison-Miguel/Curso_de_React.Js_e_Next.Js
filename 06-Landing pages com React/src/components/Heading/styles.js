import styled, { css } from 'styled-components';

const titleSize = {
    small: (theme) => css`
        font-size: ${theme.font.sizes.medium};
    `,
    medium: (theme) => css`
        font-size: ${theme.font.sizes.large};
    `,
    big: (theme) => css`
        font-size: ${theme.font.sizes.xlarge};
    `,
    huge: (theme) => css`
        font-size: ${theme.font.sizes.xhuge};
        ${mediaFont(theme)}
    `,
};

const mediaFont = (theme) => css`
    @media ${theme.media.lessThanMedium} {
        font-size: ${theme.font.sizes.xlarge};
    }
`;

const titleCase = (uppercase) => css`
    text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
    ${({ theme, colordark, size, uppercase }) => css`
        color: ${colordark ? theme.colors.primaryColor : theme.colors.white};
        ${titleSize[size](theme)}
        ${titleCase(uppercase)}
    `}
`;
