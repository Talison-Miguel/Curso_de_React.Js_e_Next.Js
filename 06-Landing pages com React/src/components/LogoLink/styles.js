import styled, { css } from 'styled-components';

export const Container = styled.a`
    ${({ theme }) => css`
        text-decoration: none;
        color: inherit;
        display: flex;
        align-items: center;

        > img {
            max-height: 3rem;
        }
    `}
`;
