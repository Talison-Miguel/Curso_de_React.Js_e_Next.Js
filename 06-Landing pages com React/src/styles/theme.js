export const theme = {
    colors: {
        primaryColor: '#0a1128',
        secondaryBg: '#dc143c',
        white: '#fff',
    },
    font: {
        family: {
            default: "'Open Sans', sans-serif",
            secondary: "'Montserrat', sans-serif",
        },
        sizes: {
            xmall: '8rem',
            small: '1.6rem',
            medium: '2.4rem',
            large: '3.2rem',
            xlarge: '4.0rem',
            xxlarge: '4.8rem',
            huge: '5.6rem',
            xhuge: '6.4rem',
        },
    },
    media: {
        lessThanMedium: '(max-width: 768px)',
    },
    spacings: {
        xmall: '8rem',
        small: '1.6rem',
        medium: '2.4rem',
        large: '3.2rem',
        xlarge: '4.0rem',
        xxlarge: '4.8rem',
        huge: '5.6rem',
        xhuge: '6.4rem',
    },
};
