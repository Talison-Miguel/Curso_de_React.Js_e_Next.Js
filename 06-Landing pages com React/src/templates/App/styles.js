import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
    ${({ theme }) => css`
        background-color: ${theme.colors.secondaryBg};
    `}
`;
