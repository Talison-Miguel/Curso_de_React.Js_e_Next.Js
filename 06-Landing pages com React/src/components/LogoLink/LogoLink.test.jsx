import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';
import { LogoLink } from '.';

import { theme } from '../../styles/theme';
import { toBeInTheDocument } from '@testing-library/jest-dom/matchers';

describe('<LogoLink />', () => {
    it('should render text logo', () => {
        renderTheme(<LogoLink link="#target" text="Olá mundo!" />);
        const heading = screen.getByRole('heading', { name: 'Olá mundo!' });
        expect(screen.getByRole('heading', { name: 'Olá mundo!' })).toBeInTheDocument();
        expect(heading.firstChild).toHaveAttribute('href', '#target');
    });

    it('should render image logo', () => {
        renderTheme(<LogoLink link="#target" text="Olá mundo!" srcImg="image.jpg" />);
        expect(screen.getByAltText('Olá mundo!')).toHaveAttribute('src', 'image.jpg');
    });

    it('should  match snapshot', () => {
        const { container } = renderTheme(<LogoLink link="#target" text="Olá mundo!" srcImg="image.jpg" />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
