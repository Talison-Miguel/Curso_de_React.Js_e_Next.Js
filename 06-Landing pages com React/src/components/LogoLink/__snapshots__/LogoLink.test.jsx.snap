// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoLink /> should  match snapshot 1`] = `
.c1 {
  text-decoration: none;
  color: inherit;
  display: flex;
  align-items: center;
}

.c1 >img {
  max-height: 3rem;
}

.c0 {
  color: #0a1128;
  font-size: 2.4rem;
  text-transform: uppercase;
}

<h1
  class="c0"
  colordark="true"
  uppercase="true"
>
  <a
    class="c1"
    href="#target"
  >
    <img
      alt="OlÃ¡ mundo!"
      src="image.jpg"
    />
  </a>
</h1>
`;
